lib/devices/device_descriptors/schwinn_x70.dart

```diff
@@ -111,26 +141,23 @@ class SchwinnX70 extends FixedLayoutDeviceDescriptor with CadenceMixin, PowerSpe
       return lastRecord ?? RecordWithSport(sport: sport);
     }
 
-    addCadenceData(time! / 1024, getCadence(data));
-    final resistance = max((resistanceMetric?.getMeasurementValue(data)?.toInt() ?? 1) - 1, 0);
-    final deltaCalories = max(calories! - lastCalories, 0);
-    lastCalories = calories;
-    var deltaTime = time - lastTime;
+    lastCalories = calories!;
+    var deltaTime = time! - lastTime;
     lastTime = time;
     if (deltaTime < 0) {
       deltaTime += 65536;
     }
 
-    // Custom way from
-    // https://github.com/ursoft/connectivity-samples/blob/main/BluetoothLeGatt/Application/src/main/java/com/example/android/bluetoothlegatt/BluetoothLeService.java
-    final power = (deltaCalories / deltaTime * 0.42 * resistancePowerFactor[resistance]);
-
+    addCadenceData(time / 1024, getCadence(data));
+    final cadence = min(computeCadence().toInt(), maxByte);
+    // the minimum legal value for resistance is 1
+    final resistance = max((getResistance(data)?.toInt() ?? 1), 1);
```

Comment:
I cannot find any reference why the resistance cannot be 0. If we force it to be minimum 1, we won't be able to distinguish if a device really supplies it or not. With FTMS bikes there's a separate whole Bluetooth characteristic which is responsible to tell the minimum, maximum, and step values: https://github.com/oesmith/gatt-xml/blob/master/org.bluetooth.characteristic.supported_resistance_level_range.xml

Reply:
Disregard, this code is Schwinn 170/270/570 specific only